src/framework/audio/internal/platform/win/wincoreaudiodriver.cpp

```diff
@@ -45,93 +59,99 @@ std::string CoreAudioDriver::name() const
     hr = pEnumerator->GetDefaultAudioEndpoint(eRender, eConsole, &pDdevice);
     pEnumerator->Release();
     CHECK_HRESULT(hr);
 
-    hr = pDdevice->Activate(IID_IAudioClient, CLSCTX_ALL,NULL, (void**)&m_audioClient);
+    hr = pDdevice->Activate(MU_IID_IAudioClient, CLSCTX_ALL,NULL, (void**)&s_data->audioClient);
     pDdevice->Release();
     CHECK_HRESULT(hr);
 
-    WAVEFORMATEX pFormat, * deviceFormat;
-    hr = m_audioClient->GetMixFormat(&deviceFormat);
+    WAVEFORMATEX* deviceFormat;
+    hr = s_data->audioClient->GetMixFormat(&deviceFormat);
     CHECK_HRESULT(hr);
 
     REFERENCE_TIME defaultTime, minimumTime;
-    hr = m_audioClient->GetDevicePeriod(&defaultTime, &minimumTime);
+    hr = s_data->audioClient->GetDevicePeriod(&defaultTime, &minimumTime);
     CHECK_HRESULT(hr);
 
-    pFormat = *deviceFormat;
-    pFormat.wFormatTag = WAVE_FORMAT_IEEE_FLOAT;
-    pFormat.nChannels = spec.channels;
-    LOGI() << pFormat.nSamplesPerSec;
-    pFormat.wBitsPerSample = 32;
-    pFormat.nAvgBytesPerSec = pFormat.nSamplesPerSec * pFormat.nChannels * sizeof(float);
-    pFormat.nBlockAlign = (pFormat.nChannels * pFormat.wBitsPerSample) / 8;
-    pFormat.cbSize = 0;
+    s_data->pFormat = *deviceFormat;
+    s_data->pFormat.wFormatTag = WAVE_FORMAT_IEEE_FLOAT;
+    s_data->pFormat.nChannels = spec.channels;
+    LOGI() << s_data->pFormat.nSamplesPerSec;
+    s_data->pFormat.wBitsPerSample = 32;
+    s_data->pFormat.nAvgBytesPerSec = s_data->pFormat.nSamplesPerSec * s_data->pFormat.nChannels * sizeof(float);
+    s_data->pFormat.nBlockAlign = (s_data->pFormat.nChannels * s_data->pFormat.wBitsPerSample) / 8;
+    s_data->pFormat.cbSize = 0;
+    s_data->callback = spec.callback;
     if (activeSpec) {
         *activeSpec = spec;
         activeSpec->format = Format::AudioF32;
-        activeSpec->sampleRate = pFormat.nSamplesPerSec;
+        activeSpec->sampleRate = s_data->pFormat.nSamplesPerSec;
     }
-    hr = m_audioClient->Initialize(AUDCLNT_SHAREMODE_SHARED, AUDCLNT_STREAMFLAGS_EVENTCALLBACK,
-                                   defaultTime, defaultTime, &pFormat, NULL);
+    hr = s_data->audioClient->Initialize(AUDCLNT_SHAREMODE_SHARED, AUDCLNT_STREAMFLAGS_EVENTCALLBACK,
+                                         defaultTime, defaultTime, &s_data->pFormat, NULL);
     CHECK_HRESULT(hr);
 
     UINT32 bufferFrameCount;
-    hr = m_audioClient->GetBufferSize(&bufferFrameCount);
+    hr = s_data->audioClient->GetBufferSize(&bufferFrameCount);
     CHECK_HRESULT(hr);
 
-    hr = m_audioClient->GetService(IID_IAudioRenderClient, (void**)&m_renderClient);
+    hr = s_data->audioClient->GetService(MU_IID_IAudioRenderClient, (void**)&s_data->renderClient);
     CHECK_HRESULT(hr);
 
-    auto hEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+    s_data->hEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
     CHECK_HRESULT(hr);
-    hr = m_audioClient->SetEventHandle(hEvent);
+    hr = s_data->audioClient->SetEventHandle(s_data->hEvent);
```

Comment:
I have a feeling that line 120 and 121 are in the wrong order, or isn't that the case?

Reply:
Exactly, thanks. I'll fix it in some other PR.